src/securechip/securechip.c

```diff
@@ -484,13 +484,25 @@ static ATCA_STATUS _update_kdf_key(void)
     UTIL_CLEANUP_32(encryption_key);
     _interface_functions->get_encryption_key(encryption_key);
 
+    uint8_t nonce_contribution[32] = {0};
+    UTIL_CLEANUP_32(nonce_contribution);
+    _interface_functions->random_32_bytes(nonce_contribution);
+#if NONCE_NUMIN_SIZE > 32
```

Comment:
`!= 32`? Or improve the error message if `< 32` is also allowed.

Reply:
We gather 32 random bytes and pass it to the cryptoauthlib function, which uses NONCE_NUMIN_SIZE bytes of that. Technically it can be less (probably never would), so I could make the check `!=32`, but it would definitely be a problem if it was more, then it would be an overflow bug.

Reply:
Got it - I guess from our side we aren't guaranteed that `NONCE_NUMIN_SIZE` is 32 exactly, so a more relaxed check is better.